### PR TITLE
Fix for disabled system repos when re-running upgrade

### PR DESCRIPTION
**Pull request by plirglo**
- Restore system repositories when re-download is needed
- Move repository scripts from /tmp to /var/tml

### CODE REVIEW

File: ansible/playbooks/roles/repository/files/download-requirements/centos-7/download-requirements.sh [hunk 1]
```diff
@@ -607,6 +607,21 @@ else
 	fi
 fi
 
+# --- Restore system repositories in case epirepo is enabled
+
+enable_system_repos_script="/var/tmp/epi-repository-setup-scripts/enable-system-repos.sh"
+disable_epirepo_client_script="/var/tmp/epi-repository-setup-scripts/disable-epirepo-client.sh"
+
+if [[ -f /etc/yum.repos.d/epirepo.repo ]]; then
+    if [[ -f /var/tmp/enabled-system-repos.txt && $enable_system_repos_script ]]; then
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/files/download-requirements/centos-7/download-requirements.sh]:
```suggestion
    if [[ -f /var/tmp/enabled-system-repos.txt && -f $enable_system_repos_script ]]; then
```

File: ansible/playbooks/roles/repository/files/download-requirements/centos-7/download-requirements.sh [hunk 2]
```diff
@@ -607,6 +607,21 @@ else
...
+        echol "OS repositories seems missing, restoring..."
+		$enable_system_repos_script || echol "Could not restore system repositories"
+		$disable_epirepo_client_script || echol "Could not disable epirepo"
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/files/download-requirements/centos-7/download-requirements.sh]:
```suggestion
		$enable_system_repos_script || exit_with_error "Could not restore system repositories"
		$disable_epirepo_client_script || exit_with_error "Could not disable epirepo"
```

File: ansible/playbooks/roles/repository/files/download-requirements/ubuntu-18.04/download-requirements.sh [hunk 3]
```diff
@@ -52,12 +52,16 @@ echol "Docker platform: $docker_platform"
...
 
-repos_backup_file="/tmp/epi-repository-setup-scripts/enable-system-repos.sh"
 # restore system repositories in case they're missing if ansible role gets interrupted
+
+enable_system_repos_script="/var/tmp/epi-repository-setup-scripts/enable-system-repos.sh"
+disable_epirepo_client_script="/var/tmp/epi-repository-setup-scripts/disable-epirepo-client.sh"
+
 if [[ ! -f /etc/apt/sources.list ]]; then
-    if [[ -f /var/tmp/enabled-system-repos.tar && -f $repos_backup_file ]]; then
+    if [[ -f /var/tmp/enabled-system-repos.tar && -f $enable_system_repos_script ]]; then
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/files/download-requirements/ubuntu-18.04/download-requirements.sh:61]: 👍🏻

File: ansible/playbooks/roles/repository/files/download-requirements/ubuntu-18.04/download-requirements.sh [hunk 4]
```diff
@@ -52,12 +52,16 @@ echol "Docker platform: $docker_platform"
...
         echol "OS repositories seems missing, restoring..."
-        $repos_backup_file
+        $enable_system_repos_script || echol "Could not restore system repositories"
+        $disable_epirepo_client_script || echol "Could not disable epirepo"
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/files/download-requirements/ubuntu-18.04/download-requirements.sh]:
```suggestion
        $enable_system_repos_script || exit_with_error "Could not restore system repositories"
        $disable_epirepo_client_script || exit_with_error "Could not disable epirepo"
```

File: ansible/playbooks/roles/repository/files/download-requirements/redhat-7/download-requirements.sh [hunk 5]
```diff
@@ -624,6 +624,21 @@ else
 	fi
 fi
 
+# --- Restore system repositories in case epirepo is enabled
+
+enable_system_repos_script="/var/tmp/epi-repository-setup-scripts/enable-system-repos.sh"
+disable_epirepo_client_script="/var/tmp/epi-repository-setup-scripts/disable-epirepo-client.sh"
+
+if [[ -f /etc/yum.repos.d/epirepo.repo ]]; then
+    if [[ -f /var/tmp/enabled-system-repos.txt && $enable_system_repos_script ]]; then
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/files/download-requirements/redhat-7/download-requirements.sh]:
```suggestion
    if [[ -f /var/tmp/enabled-system-repos.txt && -f $enable_system_repos_script ]]; then
```

File: ansible/playbooks/roles/repository/files/download-requirements/centos-7/download-requirements.sh [hunk 6]
```diff
@@ -607,6 +607,21 @@ else
...
+    else
+        echol "/var/tmp/enabled-system-repos.txt seems missing, you either know what you're doing or you need to fix your repositories"
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/files/download-requirements/centos-7/download-requirements.sh]:
```suggestion
        echol "/var/tmp/enabled-system-repos.txt or $enable_system_repos_script seems missing, you either know what you're doing or you need to fix your repositories"
```

File: ansible/playbooks/roles/repository/files/download-requirements/redhat-7/download-requirements.sh [hunk 7]
```diff
@@ -624,6 +624,21 @@ else
...
+        echol "OS repositories seems missing, restoring..."
+		$enable_system_repos_script || echol "Could not restore system repositories"
+		$disable_epirepo_client_script || echol "Could not disable epirepo"
+    else
+        echol "/var/tmp/enabled-system-repos.txt seems missing, you either know what you're doing or you need to fix your repositories"
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/files/download-requirements/redhat-7/download-requirements.sh]:
```suggestion
        echol "/var/tmp/enabled-system-repos.txt or $enable_system_repos_script seems missing, you either know what you're doing or you need to fix your repositories"
```

File: ansible/playbooks/roles/repository/files/download-requirements/redhat-7/download-requirements.sh [hunk 8]
```diff
@@ -624,6 +624,21 @@ else
...
+# --- Restore system repositories in case epirepo is enabled
+
+enable_system_repos_script="/var/tmp/epi-repository-setup-scripts/enable-system-repos.sh"
+disable_epirepo_client_script="/var/tmp/epi-repository-setup-scripts/disable-epirepo-client.sh"
+
+if [[ -f /etc/yum.repos.d/epirepo.repo ]]; then
+    if [[ -f /var/tmp/enabled-system-repos.txt && $enable_system_repos_script ]]; then
+        echol "OS repositories seems missing, restoring..."
+		$enable_system_repos_script || echol "Could not restore system repositories"
+		$disable_epirepo_client_script || echol "Could not disable epirepo"
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/files/download-requirements/redhat-7/download-requirements.sh]:
```suggestion
		$enable_system_repos_script || exit_with_error "Could not restore system repositories"
		$disable_epirepo_client_script || exit_with_error "Could not disable epirepo"
```

**Review by to-bar — CHANGES_REQUESTED** [review]
Please update paths in `docs/home/howto/REPOSITORY.md`

**Review by to-bar — APPROVED** [review]

**Review by seriva — APPROVED** [review]

**Pull request merged**